File: cpp/libjoynr/basemodel/exceptions/SubscriptionException.cpp
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2017 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
#include "joynr/exceptions/SubscriptionException.h"

namespace joynr
{

namespace exceptions
{

SubscriptionException::SubscriptionException() noexcept : JoynrRuntimeException(), subscriptionId()
{
}

SubscriptionException::SubscriptionException(const std::string& message,
                                             const std::string& subscriptionIdLocal) noexcept
        : JoynrRuntimeException(message), subscriptionId(subscriptionIdLocal)
{
}

SubscriptionException::SubscriptionException(const std::string& subscriptionIdLocal) noexcept
        : JoynrRuntimeException(subscriptionIdLocal), subscriptionId(subscriptionIdLocal)
{
}

SubscriptionException::SubscriptionException(const SubscriptionException& other)
        : JoynrRuntimeException(other), subscriptionId(other.subscriptionId)
{
}

const std::string& SubscriptionException::getSubscriptionId() const
{
    return subscriptionId;
}

void SubscriptionException::setSubscriptionId(const std::string& subscriptionIdLocal)
{
    this->subscriptionId = subscriptionIdLocal;
}

const std::string& SubscriptionException::getTypeName() const
{
    return SubscriptionException::TYPE_NAME();
}

SubscriptionException* SubscriptionException::clone() const
{
    return new SubscriptionException(const_cast<SubscriptionException&>(*this));
}

const std::string& SubscriptionException::TYPE_NAME()
{
    static const std::string TYPE_NAME = "joynr.exceptions.SubscriptionException";
    return TYPE_NAME;
}

bool SubscriptionException::operator==(const SubscriptionException& other) const
{
    return _message == other.getMessage() && subscriptionId == other.getSubscriptionId();
}

} // namespace exceptions

} // namespace joynr

File: java/javaapi/src/main/java/io/joynr/exceptions/SubscriptionException.java
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2017 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package io.joynr.exceptions;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;

public class SubscriptionException extends JoynrRuntimeException {

    private static final long serialVersionUID = 1L;

    @JsonProperty("subscriptionId")
    private final String subscriptionId;

    public String getSubscriptionId() {
        return this.subscriptionId;
    }

    /**
    * DO NOT USE
    * Constructor for deserializer
    */
    public SubscriptionException(String subscriptionId,
                                 String errorMsg,
                                 StdDeserializer<SubscriptionException> deserializer) {
        this(subscriptionId, errorMsg);
    }

    public SubscriptionException(String subscriptionId) {
        this(subscriptionId, "SubscriptionId = " + subscriptionId);
    }

    public SubscriptionException(String subscriptionId, String errorMsg) {
        super(errorMsg);
        this.subscriptionId = subscriptionId;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = super.hashCode();
        result = prime * result + ((subscriptionId == null) ? 0 : subscriptionId.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (!super.equals(obj))
            return false;
        if (getClass() != obj.getClass())
            return false;
        SubscriptionException other = (SubscriptionException) obj;
        if (subscriptionId == null) {
            if (other.subscriptionId != null)
                return false;
        } else if (!subscriptionId.equals(other.subscriptionId))
            return false;
        return true;
    }
}
